File: data/readable_format/fortran/file284.f90
PROGRAM test_zdscal
    IMPLICIT NONE
    COMPLEX*16, ALLOCATABLE :: zx(:)
    INTEGER :: i
    DOUBLE PRECISION :: da
    INTEGER :: n, incx

    ! Example test case
    n = 5
    ALLOCATE(zx(n))
    da = 2.0D0
    incx = 1

    ! Initialize zx with some values
    DO i = 1, n
        zx(i) = CMPLX(DBLE(i), DBLE(i))
    END DO

    ! Call ZDSCAL
    CALL ZDSCAL(n, da, zx, incx)

    ! Output the result
    PRINT *, 'Result:'
    DO i = 1, n
        PRINT *, 'zx(', i, ') = ', zx(i)
    END DO

    DEALLOCATE(zx)
END PROGRAM test_zdscal

SUBROUTINE ZDSCAL(N,DA,ZX,INCX)
    DOUBLE PRECISION DA
    INTEGER INCX,N
    COMPLEX*16 ZX(*)

    INTEGER I,NINCX
    INTRINSIC DCMPLX

    IF (N.LE.0 .OR. INCX.LE.0) RETURN
    IF (INCX.EQ.1) THEN
        DO I = 1,N
            ZX(I) = DCMPLX(DA,0.0D0)*ZX(I)
        END DO
    ELSE
        NINCX = N*INCX
        DO I = 1,NINCX,INCX
            ZX(I) = DCMPLX(DA,0.0D0)*ZX(I)
        END DO
    END IF
    RETURN
END SUBROUTINE ZDSCAL

File: data/readable_format/cpp/file284.cpp
#include <iostream>
#include <complex>
#include <vector>

// Define the zdscal function directly in the file
void zdscal(int n, double da, std::vector<std::complex<double>>& zx, int incx) {
    if (n <= 0 || incx <= 0) return;

    if (incx == 1) {
        for (int i = 0; i < n; ++i) {
            zx[i] *= std::complex<double>(da, 0.0);
        }
    } else {
        int nincx = n * incx;
        for (int i = 0; i < nincx; i += incx) {
            zx[i] *= std::complex<double>(da, 0.0);
        }
    }
}

void printVector(const std::vector<std::complex<double>>& v) {
    for (size_t i = 0; i < v.size(); ++i) {
        std::cout << "zx[" << i << "] = " << v[i] << std::endl;
    }
}

int main() {
    int n = 5;
    double da = 2.0;
    int incx = 1;
    std::vector<std::complex<double>> zx(n);

    // Initialize zx with some values
    for (int i = 0; i < n; ++i) {
        zx[i] = std::complex<double>(i + 1, i + 1);
    }

    // Call zdscal
    zdscal(n, da, zx, incx);

    // Output the result
    std::cout << "Result:" << std::endl;
    printVector(zx);

    return 0;
}